src/test/scala/chiselTests/BoringUtilsSpec.scala

```diff
@@ -304,6 +304,39 @@ class BoringUtilsSpec extends ChiselFlatSpec with ChiselRunners with Utils with
     log should include("Can only bore into modules that are not fully closed")
   }
 
+  it should "not create a new port when source is a port" in {
+    class Baz extends RawModule {
+      val a = IO(Output(Bool()))
+      a := DontCare
+      dontTouch(a)
+    }
+
+    class Bar extends RawModule {
+      val baz = Module(new Baz)
+    }
+
+    class Foo extends RawModule {
+      val a = IO(Output(Bool()))
+
+      val bar = Module(new Bar)
+
+      a := BoringUtils.bore(bar.baz.a)
+    }
+
+    val chirrtl = circt.stage.ChiselStage.emitCHIRRTL(new Foo)
+    matchesAndOmits(chirrtl)(
+      "module Baz",
+      "module Bar",
+      "output a_bore",
+      "connect a_bore, baz.a",
+      "module Foo",
+      "wire a_bore",
+      "connect a_bore, bar.a_bore"
+    )(
+      "connect a_bore, a"
+    )
```

Comment:
Nit:

It may be cleaner to test exactly what this is looking for only, i.e.:

```suggestion
    matchesAndOmits(chirrtl)("connect a_bore, baz.a")()
```

Reply:
I will "test exactly what this is looking for", but this suggestion actually won't work. Apparently `connect a_bore, baz.a` will also match `connect a_bore, baz.a_bore`, which is what we wanted to avoid. But I can test this with just the expected omission:

```scala
    matchesAndOmits(chirrtl)()(
      "connect a_bore, a"
    )
```